web_poet/fields.py

```diff
@@ -81,54 +83,80 @@ def __init__(self, method):
                     f"@field decorator must be used on methods, {method!r} is decorated instead"
                 )
             self.original_method = method
-            self.unbound_method = None
-            self.processors: List[Tuple[Callable, bool]] = []
+            self.name: Optional[str] = None
+            self.full_name: Optional[str] = None
 
         def __set_name__(self, owner, name):
+            self.name = name
+            self.full_name = f"{get_fq_class_name(owner)}.{name}"
```

Comment:
I *think* it's possible to have 2 classes with the same fq name, e.g. if they're defined inside a function. Would using something like `id` work here? Either `f'{id(owner).{name}`, or some id of the field object itself. The attribute won't be named `full_name`, but it might provide a more robust "id" for the field.

Reply:
We can just use `id(self)` then.